### PR TITLE
Persist data to disk

### PR DESCRIPTION
**Pull request by FenrirUnbound**
When setting values in the IMDB module, we will also attempt to persist the data onto disk. This is to help be resilient against server restarts from deployments and other mishaps.

### CODE REVIEW

File: lib/imdb.js [hunk 1]
```diff
@@ -58,15 +60,27 @@ function setValue(tableName, key, data) {
     }
 
     tables[tableName][key] = data;
+
+    if (!saveFile) {
+        return;
+    }
+
+    try {
+        fs.writeFileSync(saveFile, JSON.stringify(tables), 'utf8');
```
Comment (jer) [inline lib/imdb.js]: Is Sync strictly necessary here?

Comment (FenrirUnbound) [inline lib/imdb.js]: To maintain that the operation is synchronous. We can change this to async, since IMDB's API provides async callbacks.

**Comment by jer** [comment]
👍

**Comment by jer** [comment]
👍 beautiful

File: lib/imdb.js [hunk 2]
```diff
@@ -1,4 +1,6 @@
 'use strict';
+const fs = require('fs');
+let saveFile;
```
Comment (d2lam) [inline lib/imdb.js]: can you change the name of this variable? 
saveFile sounds like a function instead of a filename.

**Comment by nkatzman** [comment]
👍

File: index.js [hunk 3]
```diff
@@ -9,25 +9,28 @@ class MyDatastore extends Datastore {
...
+    * @param  {Object}    config            Configuration
+    * @param  {Object}    [config.db]       Object to initialize the imdb with
+    * @param  {String}    [config.filename] Filepath to read/write persistant data
     * @return {Datastore}
     */
     constructor(config) {
         super();
 
         let data = {};
+        let savefile;
```
Comment (tkyi) [inline index.js]: can you change the name of `savefile` here too?

**Pull request merged**